lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -1166,8 +1166,18 @@ namespace {
 class DeclChecker : public DeclVisitor<DeclChecker> {
 public:
   ASTContext &Ctx;
+  std::vector<AbstractFunctionDecl *> definedFunctions;
 
   explicit DeclChecker(ASTContext &ctx) : Ctx(ctx) {}
+  ~DeclChecker() {
+    for (AbstractFunctionDecl *AFD : definedFunctions) {
+      TypeChecker::typeCheckAbstractFunctionBody(AFD);
+    }
+
+    for (AbstractFunctionDecl *FD : llvm::reverse(definedFunctions)) {
+      TypeChecker::computeCaptures(FD);
```

Comment:
It should be fine to combine these two loops into one. We're no longer adding nested local functions to definedFunctions.

Reply:
This patch doesn't work as-is. I'm going to force-push at this PR and use it to kill TypeChecker instead.